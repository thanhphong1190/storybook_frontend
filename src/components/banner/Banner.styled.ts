import styled from 'styled-components'
import tw from 'twin.macro'

export const BannerSection = styled.section`
  ${tw`w-full`}
`

export const Container = styled.div`
  ${tw`flex container mx-auto relative flex-row uppercase`}
`

export const AdImage = styled.img`
  ${tw`w-full h-auto cursor-pointer`}
`

export const ContentWrapper = styled.div`
  ${tw`hidden absolute top-0 left-0 w-full h-full flex-col items-center justify-center p-4 transition-colors duration-300`}
`

export const DesktopView = styled.div`
  ${tw`w-full hidden lg:flex`}
  & {
    &:hover {
      > div {
        ${tw`flex`}
      }
    }
  }
`

export const MobileView = styled.div`
  ${tw`lg:hidden bg-cover bg-center`}
`

export const MobileContainer = styled.div`
  ${tw`flex lg:hidden flex-col px-4 py-24 justify-center items-center content-center text-center`}
`

export const Title = styled.span`
  ${tw`text-white text-4xl font-bold`}
`

export const Content = styled.span`
  ${tw`text-white text-5xl font-bold lg:mb-4 leading-10 lg:leading-none mt-2 lg:mt-0`}
`

export const ButtonWrapper = styled.div`
  ${tw`mt-12`}
`

export default BannerSection
