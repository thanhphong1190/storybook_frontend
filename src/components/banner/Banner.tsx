import React from 'react'
import { Link } from '../../services/api/types'
import { ButtonLink } from '../base'
import * as S from './Banner.styled'

export interface BannerProps {
  imageURL: string
  title: string
  content: string
  action: Link
}
export const Banner = (props: BannerProps) => {
  const {
    imageURL,
    title,
    content,
    action: { url, label }
  } = props
  return (
    <S.BannerSection>
      <S.Container>
        <S.DesktopView>
          <S.AdImage src={imageURL} />
          {/* the twin.macro have not support Color opacity feature yet. 
          We can back to fix this when they support. please check issue here https://github.com/ben-rogerson/twin.macro/issues/45 */}
          <S.ContentWrapper className='hover:bg-orange hover:bg-opacity-75'>
            <S.Title>{title}</S.Title>
            <S.Content>{content}</S.Content>
            <ButtonLink href={url} color='blue'>
              {label}
            </ButtonLink>
          </S.ContentWrapper>
        </S.DesktopView>
        <S.MobileView
          style={{
            backgroundImage: `url('${imageURL}')`
          }}
        >
          <S.MobileContainer className='bg-opacity-75 bg-orange'>
            <S.Title>{title}</S.Title>
            <S.Content>{content}</S.Content>
            <S.ButtonWrapper>
              <ButtonLink href={url} color='blue'>
                {label}
              </ButtonLink>
            </S.ButtonWrapper>
          </S.MobileContainer>
        </S.MobileView>
      </S.Container>
    </S.BannerSection>
  )
}

export default Banner
