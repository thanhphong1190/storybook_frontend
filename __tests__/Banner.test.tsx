import { Banner, BannerProps } from '../src/components/banner/Banner'
import renderer from 'react-test-renderer'

const props: BannerProps = {
  imageURL: '/example/banner-bg-example.png',
  title: 'Promotion',
  content: 'Free week of fitness',
  action: {
    label: 'Enquire now',
    url: '#'
  }
}
it('renders correctly', () => {
  const tree = renderer.create(<Banner {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
