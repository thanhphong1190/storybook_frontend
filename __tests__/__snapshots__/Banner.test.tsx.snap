// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="Bannerstyled__BannerSection-wymwyn-0 dVQukg"
>
  <div
    className="Bannerstyled__Container-wymwyn-1 bJgqCs"
  >
    <div
      className="Bannerstyled__DesktopView-wymwyn-4 ityAf"
    >
      <img
        className="Bannerstyled__AdImage-wymwyn-2 kojopC"
        src="/example/banner-bg-example.png"
      />
      <div
        className="Bannerstyled__ContentWrapper-wymwyn-3 dyITAE hover:bg-orange hover:bg-opacity-75"
      >
        <span
          className="Bannerstyled__Title-wymwyn-7 jWjdLB"
        >
          Promotion
        </span>
        <span
          className="Bannerstyled__Content-wymwyn-8 kXnLFy"
        >
          Free week of fitness
        </span>
        <a
          className="Button__ButtonLink-lxhj0a-1 eZiind"
          color="blue"
          href="#"
        >
          Enquire now
        </a>
      </div>
    </div>
    <div
      className="Bannerstyled__MobileView-wymwyn-5 kwSgDz"
      style={
        Object {
          "backgroundImage": "url('/example/banner-bg-example.png')",
        }
      }
    >
      <div
        className="Bannerstyled__MobileContainer-wymwyn-6 issnzp bg-opacity-75 bg-orange"
      >
        <span
          className="Bannerstyled__Title-wymwyn-7 jWjdLB"
        >
          Promotion
        </span>
        <span
          className="Bannerstyled__Content-wymwyn-8 kXnLFy"
        >
          Free week of fitness
        </span>
        <div
          className="Bannerstyled__ButtonWrapper-wymwyn-9 cLMRVy"
        >
          <a
            className="Button__ButtonLink-lxhj0a-1 eZiind"
            color="blue"
            href="#"
          >
            Enquire now
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
